### PR TITLE
chore(tests): update language go module & make DSL in tests more readable

### LINKED ISSUE #1302: Update `openfga/language` Go module for DSL transformer/parser and refactor usages
### Checklist

- [X] I agree to the terms within the [OpenFGA Code of Conduct](https://github.com/openfga/.github/blob/main/CODE_OF_CONDUCT.md).

### Describe the problem you'd like to have solved

A lot of our tests today use the openfga/language parser for Go to make it easier to write tests using our DSL syntax instead of the JSON model syntax. You can see a few examples of this here:

https://github.com/openfga/openfga/blob/main/internal/graph/check_test.go#L43-L55
https://github.com/openfga/openfga/blob/main/internal/graph/cached_resolver_test.go#L449-L469
https://github.com/openfga/openfga/blob/main/pkg/server/server_test.go#L182-L188

A more recent commit/release of the parser supports whitespace better, and so we can refactor these usages to have better usage of whitespace in the test files. The existing tests today had to swallow whitespace to avoid parser errors, and so they read quite cryptically relative to the other whitespace in the file.

### Describe the ideal solution

Upgrade the parser and fix any of the whitespace in the tests that use the parser to parse a DSL model string.

### Alternatives and current workarounds

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by professorabhay**
fixes #1302

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -82,6 +82,7 @@ require (
 	github.com/morikuni/aec v1.0.0 // indirect
 	github.com/opencontainers/go-digest v1.0.0 // indirect
 	github.com/opencontainers/image-spec v1.1.0-rc6 // indirect
+	github.com/openfga/language v0.0.0-20240205122832-cf938f8254bd // indirect
```
Comment (rhamzeh) [inline go.mod:85]: Is this needed? It is already referenced above as:
```
github.com/openfga/language/pkg/go v0.0.0-20240131004817-1e763f816993
```

Let's update the version there instead

`go get -u github.com/openfga/language/pkg/go`

**Review by rhamzeh — CHANGES_REQUESTED** [review]

**Comment by miparnisari** [comment]
If you search the codebase for this regex

```
^type [^\{(\s]*$
```

that should list all test cases that need to be updated.

**Comment by jon-whit** [comment]
Let's' focus on the tests in the `pkg/server` and `internal` packages. The models defined elsewhere we can follow up on.

**Comment by curfew-marathon** [comment]
Thank you for your contribution.  We are going to close this out on account of the unresolved feedback.  At this time the PR is stale.  Feel free to reopen with the requested changes.

**Pull request closed without merging**